cve_bin_tool/util.py

```diff
@@ -418,11 +418,23 @@ def decode_bom_ref(ref: str) -> ProductInfo | None:
     urn_cdx = re.compile(
         r"urn:cdx:(?P<bomSerialNumber>.*?)\/(?P<bom_version>.*?)#(?P<bom_ref>.*)"
     )
+    urn_cdx_with_purl = re.compile(
+        r"urn:cdx:(?P<bomSerialNumber>[^/]+)\/(?P<bom_version>[^#]+)#(?P<purl>pkg:[^\s]+)"
+    )
     location = "location/to/product"
-    match = urn_cbt_ext_ref.match(ref) or urn_cbt_ref.match(ref) or urn_cdx.match(ref)
+    match = (
+        urn_cdx_with_purl.match(ref)
+        or urn_cbt_ext_ref.match(ref)
+        or urn_cbt_ref.match(ref)
+        or urn_cdx.match(ref)
+    )
     if match:
         urn_dict = match.groupdict()
-        if "bom_ref" in urn_dict:  # For urn_cdx match
+        if "purl" in urn_dict:  # For urn_cdx_with_purl match
+            serialNumber = urn_dict["bomSerialNumber"]
```

Comment:
Same as above: do we need any special validation of the serial number beyond what the regex is doing?  Right now it's  basically "it doesn't contain a /" from the regex, but if there's a list of allowed characters it might make more sense to have an allowlist check here.